rocksdb_admin/admin_handler.cpp

```diff
@@ -281,6 +286,39 @@ bool SetAddressOrException(const std::string& ip,
   }
 }
 
+template <typename T>
+bool DecodeThriftStruct(const string& data, T* obj) {
+  try {
+    apache::thrift::BinarySerializer::deserialize(
+        folly::StringPiece(data.data(), data.size()), *obj);
+  } catch (const std::exception& ex) {
+    LOG(ERROR) << "Error when decoding message : " << ex.what();
+    return false;
+  }
+  return true;
+}
+
+bool DeserializeKafkaPayload(
+    const void* kafka_payload,
+    const size_t payload_len,
+    admin::KafkaOperationCode* op_code,
+    std::string* value) {
+  admin::KafkaMessagePayload msg_payload;
+
+  const std::string payload(static_cast<const char *>(kafka_payload),
```

Comment:
can we change the interface of DecodeThriftStruct() to take void* and a length to avoid make a copy of the data?